python/openinference-instrumentation/src/openinference/instrumentation/__init__.py

```diff
@@ -12,11 +33,228 @@ def suppress_tracing() -> Iterator[None]:
             # No tracing will occur within this block
             ...
     """
-    try:
-        from opentelemetry.context import _SUPPRESS_INSTRUMENTATION_KEY, attach, detach, set_value
-    except ImportError:
-        yield
+
+    def __enter__(self) -> "suppress_tracing":
+        self._token = attach(set_value(_SUPPRESS_INSTRUMENTATION_KEY, True))
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback) -> None:
+        """Raise any exception triggered within the runtime context."""
+        detach(self._token)
+        return None
+
+
+class UsingAttributes:
+    slots = [
+        "_token",
+        "_session_id",
+        "_user_id",
+        "_metadata",
+        "_tags",
+    ]
+
+    def __init__(
+        self,
+        session_id: str = "",
+        user_id: str = "",
+        metadata: Dict[str, str] = {},
+        tags: List[str] = [],
+    ) -> None:
+        if not isinstance(session_id, str):
+            raise TypeError("session_id must be a string")
+        if not isinstance(user_id, str):
+            raise TypeError("user_id must be a string")
+        if not is_dict_of(metadata, key_allowed_types=(str), value_allowed_types=(str)):
+            raise TypeError("metadata must be a dictionary with string keys and values")
+        if not is_list_of(tags, str):
+            raise TypeError("tags must be a list of strings")
```

Comment:
Definitely a good idea to do validations on the expected types here. I think as a rule of thumb, we should prefer logging a warning and not adding the attribute instead of raising an exception. The reason being that the expectation of instrumentation is that it will not break your production runtime, even if the user misconfigures it or otherwise misuses it. @anticorrelator Do you agree?